Task3/js/components/Zombie.js

```diff
@@ -0,0 +1,55 @@
+function Zombie(zombie) {
+    this.zombie = zombie || null;
+    this.zombie.health = this.zombie.health || 0;
+    this.zombie.healthCurrent = this.zombie.health;
+    this.zombie.healthPercent = 100;
+
+    this.element = document.createElement('div');
+    this.healthElement = document.createElement('div');
+    this.healthCurrentElement = document.createElement('div');
+
+    this.element.classList.add('zombie');
+    this.healthElement.classList.add('zombie-health');
+    this.healthCurrentElement.classList.add('zombie-health-current');
+
+    this.zombie.getDamage = () => {
+        if (this.zombie.healthCurrent > 0) {
+            this.zombie.healthCurrent -= HIT_DAMAGE;
+        }
+
+        this.zombie.calculateHealth();
+    };
+
+    this.zombie.calculateHealth = () => {
+        this.zombie.healthPercent = this.zombie.healthCurrent / this.zombie.health  * 100;
+
+        if (this.zombie.healthPercent < 0) {
+            this.zombie.healthPercent = 0;
+        }
+    };
+
+    this.element.addEventListener('click', () => {
+        if (this.zombie !== null) {
+            this.zombie.getDamage();
+
+            this.healthCurrentElement.style.width = `${this.zombie.healthPercent}%`;
+
+            if (this.zombie.healthCurrent <= 0) {
+                this.element.classList.add('dead');
+                this.zombie = null;
+                counter++;
+
+                setTimeout(() => {
+                    this.element.remove();
+                    numberZombie.innerText = counter + 1;
+
+                    createZombie();
```

Comment:
Zombie should not create any other zombie
this allowed only in scope of app
use events